### PR TITLE
chore: improve ci driverkit tests

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area CI

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

Adds a new test on the build-driverkit job:
* test a linux 6.2 build (that will fail until #919 is merged)

It also moves driverkit tests to use a matrix strategy.

Moreover, this adds a daily cron job that tests the build against latest kernel available on archlinux, just to make sure we catch new kmod issues asap.
It can also be triggered manually passing an optional kernel release as input string.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by FedeDP** [comment]
As written in the OP, the job is failing (i added a PR trigger just to test it): https://github.com/falcosecurity/libs/actions/runs/4241349959/jobs/7371454172

But it is working fine! :)

**Comment by hhoffstaette** [comment]
Maybe use 6.1 instead of 6.0 since it's LTS and 6.0 is already EOL?

File: .github/workflows/ci.yml
Comment (Molter73) [inline .github/workflows/ci.yml:291]: We only have one step per major release for all others, maybe we just want to keep the latest here? (at least after we merge the fix for it).

Also, we might want to consider refactoring this job to use a matrix strategy, that way we might be able to configure it in an easier manner AND run them in parallel.

File: .github/workflows/latest-kernel.yml
Comment (Molter73) [inline .github/workflows/latest-kernel.yml:1]: This is awesome! Wonder if there's some way that we could get active notifications from the scheduled job, say as notifications to a slack channel for example. If we can't get active notifications we might end up not paying attention to it and having it silently fail in the background.

File: .github/workflows/latest-kernel.yml
Comment (Molter73) [inline .github/workflows/latest-kernel.yml:30]: I don't see any `print_tags` in the inputs section, am I missing something?

File: .github/workflows/latest-kernel.yml
Comment (Molter73) [inline .github/workflows/latest-kernel.yml:43]: Could we add the `--output-probe` for this as well?

File: .github/workflows/latest-kernel.yml
Comment (FedeDP) [reply .github/workflows/latest-kernel.yml:30]: Ops wrong name!

File: .github/workflows/latest-kernel.yml
Comment (FedeDP) [reply .github/workflows/latest-kernel.yml:1]: I am not sure :/ for now, i think it's ok to check latest master CI result!

Comment (Molter73) [reply .github/workflows/latest-kernel.yml:1]: That's fine, it just feel like extra work because someone needs to actively go check the scheduled run on the `Actions` section (AFAIK the schedule event doesn't get reflected on the master branch check)

File: .github/workflows/ci.yml
Comment (FedeDP) [inline .github/workflows/ci.yml:286]: This is missing for now.
/hold until it gets uploaded :)

File: .github/workflows/latest-kernel.yml
Comment (FedeDP) [reply .github/workflows/latest-kernel.yml]: I also added `--output-probe` to ci `build-libs-driverkit` job.

File: .github/workflows/ci.yml
Comment (FedeDP) [reply .github/workflows/ci.yml]: Done!

**Review by Molter73 — DISMISSED** [review]
/lgtm

**Comment by FedeDP** [comment]
I need to drop latest commit that is just for testing :)

File: .github/workflows/ci.yml
Comment (FedeDP) [reply .github/workflows/ci.yml:286]: /unhold

**Review by Molter73 — APPROVED** [review]
/lgtm

**Review by Andreagit97 — APPROVED** [review]
/approve 
thank you!

**Pull request merged**
